lib/mix/lib/mix/tasks/deps.clean.ex

```diff
@@ -29,6 +29,14 @@ defmodule Mix.Tasks.Deps.Clean do
     Mix.Project.get!()
     {opts, apps} = OptionParser.parse!(args, strict: @switches)
 
+    config = Mix.Project.config()
+
+    Mix.Project.with_deps_lock(config, fn ->
```

Comment:
```suggestion
    Mix.Project.with_deps_lock(fn ->
```

Reply:
It seems we also clean the build path, should we put a build lock in there as well as in `mix clean`?

Reply:
Good call!